src/classes/Policy.js

```diff
@@ -174,6 +177,29 @@ class Policy {
 		}
 		return { block: false, reason: allowedOnce ? BLOCK_REASON_C2P_ALLOWED_ONCE : BLOCK_REASON_GLOBAL_UNBLOCKED };
 	}
+
+	/**
+	 * Check given url against pattern which might be a regex, or a wildcard
+	 * @param  {string} url		site url
+	 * @param  {string} pattern	regex pattern
+	 * @return {boolean}
+	 */
+	matchesWildcardOrRegex(url, pattern) {
+		let regex = escapeStringRegexp(pattern);
```

Comment:
I'm not sure if I'm missing something here with the node module you're using, but when do we actually use this value that the var `regex` is initialized with?

Reply:
Did you mean to pass the var `regex` to the first `RegExp` instantiation and the `wildcardPattern` var initialization? And if so, do you still need the try/catches?

Reply:
Yup, that's absolutely what I meant to do. I was also able to get rid of the try/catches because of the escape function you commented below. Thanks for the help.